### PR TITLE
feat: add experimentalStudio flag back

### LINKED ISSUE #23338: Reintroduce `experimentalStudio` flag
Part of the Studio epic: https://github.com/cypress-io/cypress/issues/23230

AC:

- add an `experimentalStudio` flag nested under **e2e**
- when enabled, the Cypress Studio feature should work
- should throw error for CT, informing the user Studio only works with e2e right now, and CT support will come in a future version of Cypress

## Example

Enable studio:

```ts
export default defineConfig({
  e2e: {
    experimentalStudio: true
  }
})
```

This should throw an error. Here's a similar example: https://github.com/cypress-io/cypress/pull/23104/files#r941966899

```ts
export default defineConfig({
  component: {
    experimentalStudio: true
  }
})
```

This is also not allowed:

```ts
export default defineConfig({
  experimentalStudio: true
})
```

Make sure you update the type definitions.

We have a similar pattern for experiments that only work with one testing type. An example is `experimentalSingleTabRunMode`. That PR is [here](https://github.com/cypress-io/cypress/pull/23104/). You could do a similar pattern for this one.

This is where we configure it to error for `experimentalSingleTabRunMode`:  https://github.com/cypress-io/cypress/pull/23104/files#diff-63812dbb23c9353d6d1e500239c7019912de3c33fe540d2cb58768b0b858359fR660-R665.

### PR DESCRIPTION
**Pull request by lmiller1990**
- resolves #23338

**NOTE** targeting `feat/studio-cypress-10` feature branch, not develop.

- Add back `experimentalStudio` flag.
- Only works for E2E for now.
- You can set it globally, too - this way, anyone on 9.x waiting for Studio before updating to Cypress 10 can upgrade and continue w/ Studio as-in, without changing their config.
- CT throws an informative error.

### CODE REVIEW

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -2866,10 +2866,15 @@ declare namespace Cypress {
      */
     experimentalModifyObstructiveThirdPartyCode: boolean
     /**
-     * Generate and save commands directly to your test suite by interacting with your app as an end user would.
```
Comment (lmiller1990) [inline cli/types/cypress.d.ts:2869]: This description is clearly incorrect, drive-by fix.

**Review by mike-plummer — APPROVED** [review]

**Comment by mike-plummer** [comment]
@lmiller1990 Not sure whether this would be in scope for this PR or is being tackled elsewhere - looks like there's missing i18n strings under the Project Settings view:

Screen Shot 2022-08-23 at 9 23 19 AM

**Comment by astone123** [comment]
> @lmiller1990 Not sure whether this would be in scope for this PR or is being tackled elsewhere - looks like there's missing i18n strings under the Project Settings view:
> 
> Screen Shot 2022-08-23 at 9 23 19 AM

Nice catch, Mike 👍🏻 I just added them

**Comment by lmiller1990** [comment]
Thanks for the catch @mike-plummer and thanks for adding that @astone123.

Good point on the studio icons for CT when `experimentalStudio` is root - I'll leave it to you @astone123 in https://github.com/cypress-io/cypress/issues/23337 as suggested.
